examples/diffusion/diffusion-cli.cpp

```diff
@@ -442,10 +753,20 @@ int main(int argc, char ** argv) {
     const llama_vocab * vocab            = llama_model_get_vocab(model);
     std::string         formatted_prompt = format_input_text(params.prompt, params.enable_chat_template, model);
 
-    std::vector<llama_token> input_tokens = common_tokenize(vocab, formatted_prompt,
+    std::vector<llama_token> input_tokens = common_tokenize(vocab,
+                                                            formatted_prompt,
                                                             /*add special tokens*/ true,
                                                             /*parse special*/ true);
-    int                      n_input      = input_tokens.size();
+
+    // For LLaDA models, forcefully add BOS token at the beginning. TODO: check why
+    if (arch == "llada") {
+        llama_token bos_token = llama_vocab_bos(vocab);
+        if (bos_token != LLAMA_TOKEN_NULL && (input_tokens.empty() || input_tokens[0] != bos_token)) {
+            input_tokens.insert(input_tokens.begin(), bos_token);
+        }
+    }
+
```

Comment:
This should be handled by the meta data in the GGUF model. There is a boolean field for when BOS is needed or not.